### DISCUSSION TITLE
Is the rate limiting middleware in the  graphql.php file per user or for the application?

### DISCUSSION THREAD

**Discussion by drewgallagher** [rebing/graphql-laravel#1041] [Q&A]
Is the rate limiting middleware in the graphql.php file per user or for the application?

Trying to make sure that our application is allowing enough requests for high load. Right now it's set to 250 requests/min

**Comment by mfn** [comment]
If you are using Laravels rate limiting capability, it's the same as documented at https://laravel.com/docs/10.x/rate-limiting and what you make of it. If you include the user ID, it's per user. If not, then not.

HTH

**Reply by drewgallagher** [reply]
We have it configured like so in the graphql.php file without a user id anywhere: 

```
    'schemas' => [
        'v1' => [
            'query' => $queries,
            'mutation' => $mutations,
            'middleware' => ['auth:api', 'throttle:500,1'],
            'method' => ['GET', 'POST'],
        ],
```

**Reply by mfn** [reply]
Yup, that's regular Laravel HTTP middleware and if you need more control you need should add a named rate limiter in your RouteServiceProvider and reference this in the middleware usage here.

**Reply by drewgallagher** [reply]
Oh okay good to know thanks!
